dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -697,6 +983,43 @@ def pipe(max_batch_size, input_data, device, input_layout):
     check_pipeline(generate_data(31, 13, (5, 160, 80, 3), lo=0, hi=255, dtype=np.uint8),
                    pipeline_fn=pipe, input_layout="FHWC")
 
+def test_segmentation_select_masks():
+    def get_data_source(*args, **kwargs):
+        return make_batch_select_masks(*args, **kwargs)
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=None, seed=1234)
+        with pipe:
+            polygons, vertices, selected_masks=fn.external_source(
+                num_outputs=3, device=device, source=input_data
+            )
+            out_polygons, out_vertices = fn.segmentation.select_masks(
+                selected_masks, polygons, vertices, reindex_masks=False
+            )
+        pipe.set_outputs(polygons, vertices, selected_masks, out_polygons, out_vertices)
+        return pipe
+    input_data = [get_data_source(random.randint(5, 31), vertex_ndim=2, npolygons_range=(1, 5),
+                                        nvertices_range=(3, 10)) for _ in range(13)]
+    check_pipeline(input_data, pipeline_fn=pipe, devices=["cpu"])
+
+
+def test_optical_flow():
+    if not is_of_supported():
+        raise nose.SkipTest('Optial Flow is not supported on this platform')
+
+    def pipe(max_batch_size, input_data, device, input_layout=None):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        with pipe:
+            data = fn.external_source(device=device, source=input_data, cycle=False, layout=input_layout)
+            processed = fn.optical_flow(data, device=device, output_format=4)
+        pipe.set_outputs(processed)
+        return pipe
+
+    check_pipeline(
+        generate_data(31, 13, custom_shape_generator(3, 7, 100, 100, 200, 200, 3, 3), lo=0, hi=255,
+                      dtype=np.uint8),
```

Comment:
I'm wondering, if testing the optical flow with random data is proper. On one hand, we don't really care about correctness, since we just compare pipelines. On the other hand, if the data is random, the OF operator might just output big tensor of zeros... And this might not be the good result to compare between two pipelines.

Just a food for thought

Reply:
Copied data loading from the optical flow test.